python/ray/worker.py

```diff
@@ -2340,11 +2340,16 @@ def get(object_ids):
         return values
 
 
-def put(value):
+def put(value, weakref=False):
     """Store an object in the object store.
 
+    The object may not be evicted while a reference to the returned ID exists.
+
     Args:
         value: The Python object to be stored.
+        weakref: If set, allows the object to be evicted while a reference
```

Comment:
Might want to specify that this only applies for the original reference returned by `ray.put`.